From 21fc0a293e06e180916fac7be3647a464908a147 Mon Sep 17 00:00:00 2001
From: Appana Durga Kedareswara Rao <appana.durga.rao@xilinx.com>
Date: Fri, 24 Jan 2020 20:52:32 +0530
Subject: [PATCH 1057/1852] net: axienet: Allow unaligned transfers

commit a2b804c99f05e6c9775681139ecd8233d0da4646 from
https://github.com/Xilinx/linux-xlnx.git

Replace netdev_alloc_skb_ip_align with netdev_alloc_skb.
This is needed to support all configurations of AXIDMA.
If DRE in AXIDMA is not selected, the alignment of IP will cause failures.

Signed-off-by: Appana Durga Kedareswara Rao <appana.durga.rao@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index f89234ee67e8..026f15a3262b 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -266,7 +266,7 @@ static int axienet_dma_bd_init(struct net_device *ndev)
 				      sizeof(*lp->rx_bd_v) *
 				      ((i + 1) % lp->rx_bd_num);
 
-		skb = netdev_alloc_skb_ip_align(ndev, lp->max_frm_size);
+		skb = netdev_alloc_skb(ndev, lp->max_frm_size);
 		if (!skb)
 			goto out;
 
@@ -1163,7 +1163,7 @@ static int axienet_recv(struct net_device *ndev, int budget)
 		size += length;
 		packets++;
 
-		new_skb = netdev_alloc_skb_ip_align(ndev, lp->max_frm_size);
+		new_skb = netdev_alloc_skb(ndev, lp->max_frm_size);
 		if (new_skb == NULL) {
 			dev_err(lp->dev, "No memory for new_skb\n\r");
 			break;
-- 
2.31.1

